packages/types/src/sdk/db.ts

```diff
@@ -50,7 +50,7 @@ export interface Database {
   exists(): Promise<boolean>
   checkSetup(): Promise<Nano.DocumentScope<any>>
   get<T>(id?: string): Promise<T | any>
-  remove(id?: string, rev?: string): Promise<Nano.DocumentDestroyResponse>
+  remove(id: string, rev: string): Promise<Nano.DocumentDestroyResponse>
```

Comment:
I think it would actually be better to provide an overload like the PouchDB version - as there could be a lot of other usages of the `remove(document)` syntax, they might be in the JS side of things where we wouldn't be able to find it easily.

Reply:
Pushed up the overloaded version. While it's not the approach I like, and would prefer that we had the strict id and rev requirement, it does make sense for the sake of backwards compatibility.